### DISCUSSION TITLE
Is there a launch flag to spawn a new wezterm tab?

### DISCUSSION THREAD

**Discussion by copypasteonly** [wezterm/wezterm#4854] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Gnome

### WezTerm version

20230712-072601-f4abf8fd

### Ask your question!

Hi so I've set up a keybind in Gnome to execute `wezterm` so that I could open the terminal when I wish, however, knowing that wezterm has a built in tab system is there a launch flag or command so that when there's no wezterm session it would just create a new one, and if there is one then it would just create a new wezterm tab in the same window. Thanks!

**Comment by wez** [comment]
https://wezfurlong.org/wezterm/cli/cli/spawn.html

**Reply by copypasteonly** [reply]
This is a very close solution however it doesn't open a wezterm session when there is wezterm session alive. It only works when there is atleast one wezterm session. Maybe I could write a script to detect a wezterm session and run `wezterm cli spawn` instead of just `wezterm`?

**Reply by copypasteonly** [reply, marked as answer]
Here's the script if anyone's interested!
```bash
#!/bin/bash

# Define the process name
process_name="wezterm-gui"

# Check if wezterm exists
if pgrep -x "$process_name" > /dev/null; then
    echo "Process '$process_name' exists. Executing 'wezterm cli spawn'..."
    wezterm cli spawn
else
    echo "Process '$process_name' does not exist. Executing 'wezterm'..."
    wezterm
fi
```

**Reply by wez** [reply]
Nightly builds now have `wezterm start --new-tab` and `wezterm connect --new-tab` to request a new tab rather than a new window when spawning via an existing GUI instance. The new [prefer_to_spawn_tabs](https://wezfurlong.org/wezterm/config/lua/config/prefer_to_spawn_tabs.html) option allows you to make this happen by default. [?4854](https://github.com/wez/wezterm/discussions/4854) [?4946](https://github.com/wez/wezterm/discussions/4946)
